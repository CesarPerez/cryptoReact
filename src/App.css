.App {
  text-align: center;
  height: 100vh;
  position: relative;
  background-image: url(./img/stocks.jpg);
  background-color: rgb(245 245 245 / 55%);
  background-repeat: no-repeat;
  background-size: cover;
  background-blend-mode: overlay;
  min-width: 300px;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
