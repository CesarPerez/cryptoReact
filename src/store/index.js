// import { composeWithDevTools } from "redux-devtools-extension";
import counterSlice from './reducers/counter';
import { configureStore, combineReducers } from '@reduxjs/toolkit';
import authSlice from './reducers/auth';
import notifSlice from './reducers/notif';

const allReducers = combineReducers({
  counter: counterSlice.reducer,
  auth: authSlice.reducer,
  notif: notifSlice.reducer,
});

export const store = configureStore({
  reducer: allReducers,
});

export default store;
