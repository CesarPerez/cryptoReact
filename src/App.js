import './App.css';
import Notes from './components/Notes/Notes.js';
import Header from './components/Header/Header.js';
import Crypto from './components/Crypto/Crypto.js';
import { Route, Routes } from 'react-router-dom';
import Dashboard from './components/Dashboard/Dashboard';
import Login from './components/Login/Login';
import Auth from './components/Auth/Auth';
import Notification from './components/Notification/Notification';
import Footer from './components/Footer/Footer';
import Error from './components/Error/Error';
import * as rawRoutes from './constants/routes';

function App() {
  const aux = Object.keys(rawRoutes);
  let allRoutes = [];
  aux.forEach((key, i) => {
    const newElem = rawRoutes[key];
    if (newElem.id) {
      allRoutes = [...allRoutes, newElem];
    }
  });

  return (
    <div className="App">
      <Header />
      <Notification />
      <Routes className="navbar">
        <Route key="id-1" path="/error" element={<Error type="403" />} />
        <Route
          key="id-2"
          path="/"
          element={
            <Auth>
              <Dashboard />
            </Auth>
          }
        />
        <Route
          key="id-3"
          path="/notes"
          element={
            <Auth>
              <Notes />
            </Auth>
          }
        />
        <Route key="id-4" path="/login" element={<Login />} />
        <Route key="id-5" path="*" element={<Error type="404" />} />
        {allRoutes.map((elem, i) => {
          const index = 6 + i;
          return (
            <Route
              key={'id-' + index}
              path={elem.link}
              element={
                <Auth>
                  <Crypto name={elem.id} />
                </Auth>
              }
            />
          );
        })}
      </Routes>
      <Footer />
    </div>
  );
}

export default App;
