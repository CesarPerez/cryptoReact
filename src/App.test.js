import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer'; // ES6
import App from './App.js';
import store from './store';

test('Should match the snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
