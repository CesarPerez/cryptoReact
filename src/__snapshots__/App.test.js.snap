// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="navbar"
  >
    <h4
      className="brand"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        CryptoTool
      </a>
    </h4>
  </header>
  <footer
    className="containerFooter"
  >
    <div>
      <div>
        <span>
          Â© 2022 Sisar
        </span>
      </div>
    </div>
  </footer>
</div>
`;
